giturlparse/parser.py

```diff
@@ -99,6 +118,22 @@ def parse(self):
             d['user'] = m2.group(1)
             d['name'] = m2.group(6)
             d['owner'] = m2.group(5)
+        elif m3:
+            d['pathname'] = self._get_path()
+            d['protocol'] = m3.group('protocol')
+            d['user'] = m3.group('user')
+            d['resource'] = m3.group('site') + '.com'
```

Comment:
This doesn't seem ideal.  The domain '.com' is hard coded here.  It wont match any other domain.

Reply:
Ah, actually all the valid and invalid urls were for the `.com` so I hard coded this for `.com`

Reply:
GitHub and GitLab can be self-hosted, and may not be on `.com`, and there are other git hosting software.

Reply:
Yes, I have changed the regex for all the host now. :)